tests/e2e/hcp_machine_pool_test.go

```diff
@@ -1,23 +1,126 @@
 package e2e
 
 import (
+	"fmt"
+	"strconv"
 	"strings"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"github.com/openshift-online/ocm-common/pkg/test/vpc_client"
 
 	"github.com/openshift/rosa/tests/ci/labels"
+	"github.com/openshift/rosa/tests/utils/common"
+	"github.com/openshift/rosa/tests/utils/common/constants"
+	"github.com/openshift/rosa/tests/utils/config"
 	"github.com/openshift/rosa/tests/utils/exec/rosacli"
 	ph "github.com/openshift/rosa/tests/utils/profilehandler"
 )
 
-var _ = Describe("Create Machine Pool", labels.Feature.Machinepool, func() {
+var _ = Describe("HCP Machine Pool", labels.Feature.Machinepool, func() {
+	// It doesn't check whether node pool instances ready in default.
+	// If needed for verify hcp node pool's changes, pls set the ENV CLUSTER_NODE_POOL_GLOBAL_CHECK to true,
+	// which will wait for node pool instances ready until timeout.
+	isNodePoolGlobalCheck := config.IsNodePoolGlobalCheck()
 
 	var (
-		rosaClient *rosacli.Client
-		profile    *ph.Profile
+		rosaClient  *rosacli.Client
+		profile     *ph.Profile
+		isMultiArch bool
 	)
+
+	getAutoScaleReplicas := func(mpName string) (map[string]int, error) {
```

Comment:
I think that those methods could be part of the machine pool service interface. wdyt ?
Methods should return error and the error would be check in tests.
They could be reused in other tests as well if needed. And that way we don't overload too much the test files

Reply:
Yeah. Good catch. I'll adjust `getAutoScaleReplicas` to  the interface `MachinePoolService`.